src/ekka_rlog_tab.erl

```diff
@@ -0,0 +1,69 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% access module for transaction logs
+%% implemented as a ram_copies mnesia table
+-module(ekka_rlog_tab).
+
+%% Mnesia bootstrap
+-export([mnesia/1]).
+
+-export([write/3]).
+
+-include("ekka_rlog.hrl").
+-include_lib("snabbkaffe/include/trace.hrl").
+
+-type key() :: ekka_rlog_lib:txid().
+-type shard() :: ekka_rlog:shard().
+
+-boot_mnesia({mnesia, [boot]}).
+-copy_mnesia({mnesia, [copy]}).
+
+%% @doc Mnesia bootstrap.
+mnesia(BootType) ->
+    {ok, _} = ekka_mnesia_null_storage:register(),
+    case ekka_rlog:role() of
+        core -> [init(BootType, Shard) || Shard <- ekka_rlog_config:shards()], ok;
+        _    -> ok
+    end.
+
+%% @doc Write a transaction log.
+-spec write(ekka_rlog:shard(), ekka_rlog_lib:txid(), [ekka_rlog_lib:op(),...]) -> ok.
+write(Shard, Key, [_ | _] = Ops) ->
+    Log = #rlog{ key = Key
+               , ops = Ops
+               },
+    mnesia:write(Shard, Log, write).
+
+init(boot, Shard) ->
+    Opts = [ {type, ordered_set}
```

Comment:
Should this table have `{local_content, true}`?

The table will be empty, so copying the table is a noop, but without the local content, the writes will probably be distributed over the core nodes. But each node will generate its own writes, that probably are identical. Are we bouncing these writes between the nodes?

Reply:
I am not sure how table events will work for `local_content` tables. It's important that the agents receive table events from all core nodes.